source/slang/core.meta.slang

```diff
@@ -1720,6 +1720,57 @@ T __attachToNativeRef(NativeRef<T> nativeVal)
 }
 
 //@public:
+__generic<T, let N:int>
+__magic_type(ArrayExpressionType)
+struct Array : IRWArray<T>
+{
+    __intrinsic_op($(kIROp_GetArrayLength))
+    int getCount();
+}
+
+//@public:
+__generic<int N>
+__magic_type(StringLiteralExpressionType)
+__intrinsic_type($(kIROp_StringLiteralType))
+struct StringLiteral : IArray<uint>
+{
+    __intrinsic_op($(kIROp_GetStringLiteralLength))
+    [__unsafeForceInlineEarly]
+    int getCount();
+
+    property uint length
+    {
+        [__unsafeForceInlineEarly] get { return (uint)getCount(); }
+    }
+
+    __intrinsic_op($(kIROp_GetStringLiteralAsArray))
+    [__unsafeForceInlineEarly]
+    Array<uint, N> getArray();
+
+    [__unsafeForceInlineEarly]
+    uint _getElementFromArray(uint i)
```

Comment:
should be marked as `internal` so it is not callable from outside the core module.